obnova_hesla_bp.py

```diff
@@ -1,14 +1,72 @@
-from flask import Blueprint, render_template
+from flask import Blueprint, render_template, request, redirect, flash, abort, url_for, g
+from flask_login import current_user, logout_user
+import os
+import db_funkce
+from cryptography.fernet import Fernet, InvalidToken
+import json
+import posta_funkce
+
 blueprint = Blueprint('obnova_hesla_bp', __name__)
 @blueprint.route('/obnovahesla')
 def show_obnova():
 	return render_template('obnova_hesla.html')
 
-@blueprint.route('/link')
-def send_link():
-	# pokus o obnovu hesla
-	poslano = True
 
+f = Fernet(os.environ["PASSWORD_RESET_KEY"].encode("ascii"))
+@blueprint.route('/obnovahesla', methods=['POST'])
+def find_user():
+	email = request.form.get('email')
+	uzivatel = db_funkce.najdi_uzivatele(email)
+	if uzivatel:
+		flash ('Na zadaný e-mail jsme poslali link pro obnovu hesla.', "success")
+		id = uzivatel.db_id
+		token_data = {"id": id}
+		token = f.encrypt(json.dumps(token_data).encode("ascii")).decode("ascii")
+		password_reset_url = url_for('.show_heslonew', token=token, _external=True)
+		print(password_reset_url)
+		posta_funkce.email_reset_hesla(uzivatel, password_reset_url)
+		return render_template('obnova_hesla.html')
+	else:
+		flash ('Tento e-mail v naší databázi není.', "danger")
+		return render_template('obnova_hesla.html')
+
+@blueprint.route('/noveheslo/<token>')
+def show_heslonew(token):
+	if current_user.is_authenticated:
+		logout_user()
+		return redirect(url_for('.show_heslonew', token=token))
+	else:
+		try:
+			token_data = json.loads(f.decrypt(token.encode("ascii"), ttl=60*60*24))
+			#print(token_data)
+		except InvalidToken:
+			print("Neplatný token.")
+			flash ("Neplatný link.", "danger")
+		return render_template('noveheslo.html', token=token)
+
+
+@blueprint.route('/noveheslo', methods=['POST'])
+def heslo_new():
+	result = request.form
+	chyba = None
+	token = result.get("token")
+	id_uzivatele = json.loads(f.decrypt(token.encode("ascii"), ttl=60*60*24))['id']
+	heslo = result.get("heslo")
+	heslo_potvrzeni = result.get("heslo_potvrzeni")
+	if not heslo == heslo_potvrzeni:
+		flash ('Hesla se neshodují.', "danger")
+		return render_template("noveheslo.html")
+	else:
+		zmeneno = db_funkce.zmena_hesla(heslo, id_uzivatele)
+
+	if zmeneno == True:
+		flash ('Heslo bylo úspěšně změněno. Nyní se můžeš přihlásit.', "success")
+		return render_template('prihlaseni.html')
```

Comment:
Tady by byl lepší `redirect` namísto `render_template`. Rozdíl je v tomhle:
* když dáš `return render_template('prihlaseni.html')`, tak uživatel sice uvidí přihlašovací stránku, ale nahoře bude mít v browseru jako adresu `/noveheslo`.
* když dáš `return redirect(url_for('prihlaseni_bp.login'))`, bude se to chovat správně a bude zobrazená správná adresa

Mimochodem tahle poznámka se týká i několika dalších míst aplikace, co jsem tak zaznamenal.